<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PSOParameters</class>
 <widget class="QWidget" name="PSOParameters">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>478</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="psoTab">
      <attribute name="title">
       <string>粒子群优化算法</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <property name="rowWrapPolicy">
          <enum>QFormLayout::DontWrapRows</enum>
         </property>
         <property name="labelAlignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="verticalSpacing">
          <number>12</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="pLabel">
           <property name="text">
            <string>种群大小(&amp;P):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditGroup</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="LineEditGroup">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="gLabel">
           <property name="text">
            <string>迭代次数(&amp;G):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditGeneration</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="LineEditGeneration">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="wLabel">
           <property name="text">
            <string>惯性权重(&amp;W):</string>
           </property>
           <property name="buddy">
            <cstring>wWidget</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QWidget" name="wWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>20</height>
            </size>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>131</width>
              <height>20</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QSlider" name="horizontalSliderWeight">
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="labelWeight">
               <property name="text">
                <string>0.5</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="c1C1Label">
           <property name="text">
            <string>加速系数C1(&amp;C1):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditSpeed1</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="LineEditSpeed1">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="c2C2Label">
           <property name="text">
            <string>加速系数C2(&amp;C2):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditSpeed2</cstring>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="LineEditSpeed2">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="uLabel">
           <property name="text">
            <string>变量上界(&amp;U):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditUp</cstring>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QLineEdit" name="LineEditUp">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="bLabel">
           <property name="text">
            <string>变量下界(&amp;B):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditDown</cstring>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QLineEdit" name="LineEditDown">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="dLabel">
           <property name="text">
            <string>变量维度(&amp;D):</string>
           </property>
           <property name="buddy">
            <cstring>ComboBoxDimension</cstring>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QComboBox" name="ComboBoxDimension">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>20</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>30</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>50</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>100</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QTextEdit" name="textEdit">
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; -qt-user-state:0;&quot;&gt;种群大小 = 变量维度 * 20左右;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;惯性权重可以采用模糊控制的方式设定，或者在[0.5, 1.0]之间随机取值。惯性权重设为0.729的同时将c1和c2设1.49445，有利于算法的收敛;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;加速系数c1和c2通常都等于2.0，代表着对两个引导方向的同等重视;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;迭代次数设定为500，1000，5000，或者最大的函数评估次数，等等。&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="deTab">
      <attribute name="title">
       <string>差分进化算法</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayout_2">
         <property name="labelAlignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="verticalSpacing">
          <number>12</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="pLabel_2">
           <property name="text">
            <string>种群大小(&amp;P):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditGroup_D</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="LineEditGroup_D">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="gLabel_2">
           <property name="text">
            <string>迭代次数(&amp;G):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditGeneration_D</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="LineEditGeneration_D">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="fLabel">
           <property name="text">
            <string>缩放因子(&amp;F):</string>
           </property>
           <property name="buddy">
            <cstring>fWidget</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QWidget" name="fWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>181</width>
              <height>20</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QSlider" name="horizontalSliderFactor_D">
               <property name="maximum">
                <number>200</number>
               </property>
               <property name="singleStep">
                <number>2</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="labelFactor_D">
               <property name="text">
                <string>0.5</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="cRLabel">
           <property name="text">
            <string>交叉概率(&amp;CR):</string>
           </property>
           <property name="buddy">
            <cstring>cRWidget</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QWidget" name="cRWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>121</width>
              <height>20</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QSlider" name="horizontalSliderCR_D">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="labelCR_D">
               <property name="text">
                <string>0.3</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="uLabel_2">
           <property name="text">
            <string>变量上界(&amp;U):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditUp_D</cstring>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="LineEditUp_D">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="bLabel_2">
           <property name="text">
            <string>变量下界(&amp;B):</string>
           </property>
           <property name="buddy">
            <cstring>LineEditDown_D</cstring>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QLineEdit" name="LineEditDown_D">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="dLabel_2">
           <property name="text">
            <string>变量维度(&amp;D):</string>
           </property>
           <property name="buddy">
            <cstring>ComboBoxDimension_D</cstring>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="ComboBoxDimension_D">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>20</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>30</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>50</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>100</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QTextEdit" name="textEdit_2">
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; -qt-user-state:0;&quot;&gt;种群大小：一般介于5×维度 与10×维度 之间, 但不能少于4, 否则无法进行变异操作；&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;缩放因子：一般在[ 0, 2 ]之间选择, 通常取0. 5；&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;交叉概率：一般在[ 0, 1 ]之间选择, 比较好的选择应在0. 3 左右, CR大些收敛速度会加快, 但易发生早熟现象。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>迭代进度：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>当前函数(F):</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBoxFunction">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string>bent_cigar_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>sum_diff_pow_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>zakharov_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>rosenbrock_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>rastrigin_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>schaffer_F7_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>bi_rastrigin_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>step_rastrigin_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>levy_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>schwefel_func</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf01</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf02</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf03</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf04</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf05</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf06</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf07</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf08</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf09</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hf10</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf01</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf02</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf03</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf04</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf05</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf06</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf07</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf08</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf09</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>cf10</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>最优结果(R):</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEditResult">
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPushButton" name="pushButtonProcess">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>50</width>
       <height>60</height>
      </size>
     </property>
     <property name="text">
      <string>运行</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
